### PR TITLE
fix(k8s): stabilize K8S functional tests

### PR DESCRIPTION
**Pull request by vponomaryov**
Do following changing to the K8S functional tests:
- `test_deploy_helm_with_default_values`
    Use selector for scylla pods filtering out the cleanup ones.
    Catch namespace deletion timeouts and suppress raising error.
    Enable this test for ARM setup.
- `test_scylla_yaml_override`
    Add 5s sleeps between configmap updates and triggers of rollout restart.
    It will allow us to avoid concurrency when configmap is not yet updated at rollout trigger moment.

### Testing

- [ ]

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
Failures (which get fixed in this PR) examples:
- `test_scylla_yaml_override` flakiness: [scylla-operator/operator-1.12/functional/functional-eks-test#6](https://argus.scylladb.com/test/f15e1983-30f7-4235-a387-9d2e4f55508c/runs?additionalRuns[]=a98076b1-e4dd-48a2-ba4a-421c642c20dd)
- `test_deploy_helm_with_default_values`.
-- Example of the problem without proper selector usage: [scylla-operator/operator-1.12/functional/functional-k8s-local-kind-aws-test#7](https://argus.scylladb.com/test/3e750c0e-6720-4973-bb0b-3862ca4918a0/runs?additionalRuns[]=e7513e4d-0ae1-423c-8d65-e4d62de33d44)
-- Example of the `namespace deletion timeout` problem: [scylla-operator/operator-master/functional/functional-k8s-local-kind-aws-test#11](https://argus.scylladb.com/test/bbc6b4d9-cc3e-4b87-8e29-8d1b9381f091/runs?additionalRuns[]=7828022c-bb0b-441f-82ba-52353301e23f)

Successful test runs with this fix:
- [scylla-staging/valerii/vp-functional-eks#73](https://argus.scylladb.com/test/e9294e61-0380-4232-9945-04f8d9837037/runs?additionalRuns[]=3808c50d-62ca-41d8-9ec9-a7c2326b4730)
- [scylla-staging/valerii/vp-functional-eks-arm#16](https://argus.scylladb.com/test/58add5d0-7dc8-4e04-9738-7e03fb5dcb30/runs?additionalRuns[]=731e2c0f-51a7-4355-aeeb-09e9fde1eb68)
- [scylla-staging/valerii/vp-functional-k8s-local-kind-aws#325](https://argus.scylladb.com/test/9e6ab4ee-3fcf-406a-b359-f583c4da5351/runs?additionalRuns[]=82e55aff-07e1-4818-97d7-76c8a83b57ca)
- [scylla-staging/valerii/vp-functional-k8s-local-kind-aws#326](https://argus.scylladb.com/test/9e6ab4ee-3fcf-406a-b359-f583c4da5351/runs?additionalRuns[]=379abe18-0e9b-4865-ab3c-bfb79293f46b)

File: functional_tests/scylla_operator/test_functional.py [hunk 1]
```diff
@@ -896,6 +891,7 @@ def test_scylla_yaml_override(db_cluster, scylla_yaml):  # pylint: disable=too-m
         else:
             dict(props).pop(hh_throttle_option_name)
 
+    time.sleep(5)
```
Comment (soyacz) [inline functional_tests/scylla_operator/test_functional.py:894]: worth to add a comment line to such sleeps, so it's clear without using git-blame

**Review by soyacz — APPROVED** [review]
LGTM

**Review by fruch — APPROVED** [review]
LGTM

File: functional_tests/scylla_operator/test_functional.py (hunk 1 above)
Comment (vponomaryov) [reply functional_tests/scylla_operator/test_functional.py:896]: Done, added comment.

**Comment by vponomaryov** [comment]
@fruch it is waiting for you.

**Pull request merged**